File: replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int max=arr[arr.length-1];
        for(int i = arr.length-1; i>=0; i--){
            if(i==arr.length-1){
                arr[i] = -1;
            }else{
                int temp = arr[i];
                arr[i]=max;
                if(temp>max){
                    max=temp;
                }
            }
        }
        return arr;
    }
}

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        ans = [ 0 for num in range(len(arr))]
        m = -1
        for index in range(len(arr) - 1, -1, -1):
            if index == len(arr) - 1:
                ans[index] = m
                m = arr[index]
                continue
            
            ans[index] = m

            m = max(arr[index], m)

        return ans